src/py42/services/legalholdapiclient.py

```diff
@@ -0,0 +1,373 @@
+from py42 import settings
+from py42.exceptions import Py42BadRequestError
+from py42.exceptions import Py42Error
+from py42.exceptions import Py42ForbiddenError
+from py42.exceptions import Py42LegalHoldAlreadyActiveError
+from py42.exceptions import Py42LegalHoldAlreadyDeactivatedError
+from py42.exceptions import Py42LegalHoldCriteriaMissingError
+from py42.exceptions import Py42LegalHoldNotFoundOrPermissionDeniedError
+from py42.exceptions import Py42UserAlreadyAddedError
+from py42.services import BaseService
+from py42.services.util import get_all_pages
+
+
+def _active_state_map(active):
+    _map = {True: "ACTIVE", False: "INACTIVE", None: "ALL"}
+    try:
+        return _map[active]
+    except KeyError:
+        raise Py42Error(
+            f"Invalid argument: '{active}'. active must be True, False, or None"
+        )
+
+
+class LegalHoldApiClientService(BaseService):
```

Comment:
It may be nicer to stick this into the `services/legalhold.py` file to prevent some redundancies and not have these wordy file names - but this follows our convention of a file for each service.

Reply:
I generally err on the side of less scrolling